Models/global_config.hpp

```diff
@@ -0,0 +1,128 @@
+#ifndef ZPSW3_GLOBAL_CONFIG_HPP
+#define ZPSW3_GLOBAL_CONFIG_HPP
+
+#include <cstdint>
+
+#include "AM_ControlAlgorithm.hpp"
+#include "ZP_D_PWMChannel.hpp"
+
+namespace config
+{
+    using percentage_t = float;
+
+        class TempDSHOTDriver : public MotorChannel
+        {
+            public:
+                TempDSHOTDriver() {}
+                void set(uint8_t percent){}
+        };
+    class RCInputDriver{};
+        class TempPPMDriver : public RCInputDriver{};
+        class TempSBusDriver : public RCInputDriver{};
+    class UnifiedSensorDriver{};
+        class TempVN300Driver : public UnifiedSensorDriver{};
+        class TempVN310Driver : public UnifiedSensorDriver{};
+    class GPSDriver{};
+        class TempNEOM8Driver : public GPSDriver{};
+        class otherGPSDriver : public GPSDriver{};
+    class IMUDriver{};
+        class TempMPU6050Driver : public IMUDriver{};
+    class AirspeedDriver{};
+        class someAirspeedDriver : public AirspeedDriver{};
+     
+    //Define factory function to create drivers
+    template <class BaseDriverType, class DerivedDriverType, auto... args>
+    BaseDriverType* constructDriver() {
+        return new DerivedDriverType(args...);
+    }
+
+    //Typedef pointer to factory function
+    template <class BaseDriverType>
+    using DriverFactory = BaseDriverType* (*)(void);
+
+
+
+    /* Motor declarations */
+
+    typedef enum {
+        yaw,
+        pitch,
+        roll,
+        thrust
+    } ControlAxis_t;
+
+    typedef struct {
+        ControlAxis_t axis;
+        bool isInverted = false;
+        DriverFactory<MotorChannel> driverConstructor;
+    } Motor_t;
+
+
+
+    /* RC input declarations */
+
+    typedef struct {
+        //TODO: determine other config fields relevant to the RC input type
+        DriverFactory<RCInputDriver> driverConstructor;
+    } RCInput_t;
+
+
+
+    /* GPS declarations */
+
+    typedef struct {
+        DriverFactory<GPSDriver> driverConstructor;
+    } GPS_t;
+
+
+
+    /* Flightmode declarations */
+
+    //Define factory function to create flightmodes
+    template <class FlightmodeType, auto... args>
+    AM::Flightmode* constructFlightmode() {
+        return new FlightmodeType(args...);
+    }
```

Comment:
This seems functionally equivalent to `constructDriver`?

Reply:
Would it be better then to just use the same function template as the drivers, passing Flightmode and ExampleFlightmode1 as BaseDriverType and DerivedDriverType?

Reply:
I think so, and we can just use a more generic name for the function that applies to all scenarios.